extensions/big-data-cluster/src/bigDataCluster/tree/controllerTreeNode.ts

```diff
@@ -0,0 +1,361 @@
+	}
+
+	private getEndPointFolderNode(): FolderNode {
+		let label = 'SQL Servers';
+		let epFolderNode = this.children.find(e => e instanceof FolderNode && e.label === label);
+		if (!epFolderNode) {
+			epFolderNode = new FolderNode({ label, parent: this, treeChangeHandler: this.treeChangeHandler });
+			this.addChild(epFolderNode);
+		}
+		return epFolderNode as FolderNode;
+	}
+
+	public getTreeItem(): vscode.TreeItem {
+		let item: vscode.TreeItem = super.getTreeItem();
+		item.collapsibleState = vscode.TreeItemCollapsibleState.Collapsed;
+		return item;
+	}
+
+	public get url() {
+		return this._url;
+	}
+
+	public set url(url: string) {
+		this._url = url;
+	}
+
+	public get username() {
+		return this._username;
+	}
+
+	public set username(username: string) {
+		this._username = username;
+	}
+
+	public get password() {
+		return this._password;
+	}
+
+	public set password(pw: string) {
+		this._password = pw;
+	}
+
+	public get rememberPassword() {
+		return this._rememberPassword;
+	}
+
+	public set rememberPassword(rememberPassword: boolean) {
+		this._rememberPassword = rememberPassword;
+	}
+}
+
+export class FolderNode extends ControllerTreeNode {
+	constructor(properties?: {
+		label: string;
+		parent: ControllerTreeNode;
+		treeChangeHandler: IControllerTreeChangeHandler;
+	}) {
+		super(Object.assign({
+			description: properties.label,
+			nodeType: BdcItemType.folder,
+			iconPath: IconPath.folderNode
+		}, properties));
+	}
+}
+
+export class SqlMasterNode extends ControllerTreeNode {
+	private _role: string;
+	private _endPointAddress: string;
+	private _username: string;
+	private _password: string;
+
+	constructor(properties?: {
+		endPointAddress: string,
+		parent: ControllerTreeNode,
+		treeChangeHandler: IControllerTreeChangeHandler,
+		label?: string,
+		description?: string,
+	}) {
+		super(Object.assign({
+			label: undefined,
+			nodeType: BdcItemType.sqlMaster,
+			iconPath: IconPath.sqlMasterNode
+		}, properties));
+
+		this._endPointAddress = properties.endPointAddress;
+		this._username = 'sa';
+		this._role = 'sql-server-master';
+		this.label = properties.label || `master: ${this._endPointAddress} (${this._username})`;
+		this.description = properties.description || this.label;
+	}
+
+	private getPassword(): string {
+		if (!this._password) {
+			let current: TreeNode = this;
+			while (current && !(current instanceof ControllerNode)) {
+				current = current.parent;
+			}
+			this._password = current && current instanceof ControllerNode ? current.password : undefined;
+		}
+		return this._password;
+	}
+
+	public getTreeItem(): vscode.TreeItem {
+		let item = super.getTreeItem();
+		let connectionProfile: azdata.IConnectionProfile = {
+			id: this.id,
+			connectionName: this.id,
+			serverName: this._endPointAddress,
+			databaseName: '',
+			userName: this._username,
+			password: this.getPassword(),
+			authenticationType: 'SqlLogin',
```

Comment:
Do we not support other types of auth for connecting to the nodes?

Reply:
Yes. That is current requirement. We would add more way of auth in near future.